### PR TITLE
Update documentation for Larq Compute Engine 0.4

### PR DESCRIPTION
**Pull request by lgeiger**
This PR adds some initial documentation for the LCE operators.

It is marked as a draft PR as it documents the options used on `master` and not the ones in our current stable release.

**A rendered preview of can be found at https://nightly.docs.larq.dev/compute-engine/**

### CODE REVIEW

File: docs/compute-engine/api/operators.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+\\]
+
+where \\(\star\\) is the 2D cross-correlation operator, \\(N\\) is a batch size, \\(I\\) and \\(O\\) denote the number of input and output channels, and \\(\mathrm{bsign}\\) is the [binary sign function](/larq/api/math/#sign-function).
+
+**Inputs**
+
+- `Tensor<float32|int8>|BitTensor<int32,3>`: 4D input tensor \\(x\\)
+- `BitTensor<int32,3>`: 4D bitpacked binary filter tensor \\(w\\) in`OHWI` format
+- `Tensor<float32>`: Post activation multiplier \\(\gamma\\)
+- `Tensor<float32>`: Post activation bias \\(\beta\\)
```
Comment (Tombana) [inline docs/compute-engine/api/operators.md:30]:
```suggestion
- `Tensor<float32>`: 1D post activation multiplier \\(\gamma\\)
- `Tensor<float32>`: 1D post activation bias \\(\beta\\)
```
(and maybe we should add that `filter.dim(0) == post_multiplier.dim(0) == post_bias.dim(0)` but i'm not sure how to specify that cleanly)

**Comment by koenhelwegen** [comment]
I would avoid `o` as an index symbol, would use either `out` or just `j`.

**Comment by lgeiger** [comment]
> I would avoid o as an index symbol, would use either out or just j.

Why, do you think `o` looks confusing? I originally had long names but this somehow broke the LaTeX parser and then went for a naming scheme similar to a physics book. I didn't go with `j` because now we would have a direct relation of the indices to the `OHWI` dataformat used for the filters. But I'll check and see if I can get the longer names to show up nicely

**Comment by koenhelwegen** [comment]
> Why, do you think `o` looks confusing? I originally had long names but this somehow broke the LaTeX parser and then went for a naming scheme similar to a physics book. I didn't go with `j` because now we would have a direct relation of the indices to the `OHWI` dataformat used for the filters. But I'll check and see if I can get the longer names to show up nicely

Yes this is a lot better I think! The `o` was a bit too close to a 0 for my taste.

**Comment by lgeiger** [comment]
> The o was a bit too close to a 0 for my taste.

Makes sense :+1:

File: docs/compute-engine/api_page_templates/operators.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+The final output with type `Tensor<float32|int8>` is then calculated as
+\\[
+y_{n,\mathrm{out}} = \beta_\mathrm{out} + \gamma_\mathrm{out} \, \sigma\left(\hat{y}\_{n,\mathrm{out}}\right)\text{.}
+\\]
+If the output type is `BitTensor<int32, 3>` the final transformation is simplified to
+\\[
+y_{n,\mathrm{out}} = \begin{cases}
+  -1.0 & \hat{y}\_{n,\mathrm{out}} < \tau_\mathrm{out} \\\
+  \hphantom{-}1.0 & \hat{y}\_{n,\mathrm{out}} \geq \tau_\mathrm{out}\text{.}
```
Comment (AdamHillier) [inline docs/compute-engine/api_page_templates/operators.md:31]: I think it should be:

```suggestion
  -1.0 & \hat{y}\_{n,\mathrm{out}} > \tau_\mathrm{out} \\\
  \hphantom{-}1.0 & \hat{y}\_{n,\mathrm{out}} \leq \tau_\mathrm{out}\text{.}
```

**Pull request merged**